src/shared/components/query/MolecularProfileSelector.tsx

```diff
@@ -21,6 +21,27 @@ export default class MolecularProfileSelector extends QueryStoreComponent<
         return getServerConfig().skin_show_gsva;
     }
 
+    componentDidMount() {
+        // Select RNA expression profile by default if no mutation data is present
+        const hasMutationProfile =
+            this.store.getFilteredProfiles(
+                'MUTATION_EXTENDED' as MolecularProfile['molecularAlterationType']
+            ).length > 0;
+        const hasMrnaProfile =
+            this.store.getFilteredProfiles(
+                'MRNA_EXPRESSION' as MolecularProfile['molecularAlterationType']
+            ).length > 0;
+
+        if (!hasMutationProfile && hasMrnaProfile) {
+            const mrnaProfiles = this.store.getFilteredProfiles(
+                'MRNA_EXPRESSION' as MolecularProfile['molecularAlterationType']
+            );
+            if (mrnaProfiles.length > 0) {
+                this.store.selectMolecularProfile(mrnaProfiles[0], true);
+            }
+        }
+    }
```

Comment:
@DavidZ79 if there's an easy way to do this in the render method instead that would be better,  but no worries if not

Reply:
Thanks @inodb and @alisman for the feedback, I moved the default mRNA profile selection logic out of componentDidMount and into QueryStore. Let me know if this is good or you'd like me to tweak it.